### PR TITLE
allow one to check if mech has been instantiated by checking alloc function

### PR DESCRIPTION
**Pull request by brunomaga**

### CODE REVIEW

File: coreneuron/nrnoc/register_mech.cpp [hunk 1]
```diff
@@ -159,6 +159,7 @@ void alloc_mech(int n) {
         int i;
         for (i = 0; i < memb_func_size_; ++i) {
             nrn_mech_data_layout_[i] = 1;
+            memb_func[i].alloc = NULL; //mark as non-instantiated
```
Comment (pramodk) [inline coreneuron/nrnoc/register_mech.cpp]: `members_func` is allocated with `calloc` so it's already initialised to 0 (unless I am missing something here...)

Comment (brunomaga) [reply coreneuron/nrnoc/register_mech.cpp]: @pramodk this commit fixes verification of the instantiation for non-`calloc` allocation of `memb_func`

Comment (pramodk) [reply coreneuron/nrnoc/register_mech.cpp]: could you check why build was failed? might need close and reopen.

**Comment by pramodk** [comment]
`alloc_mech` allocates with calloc so I don't see how this is relevant here now.

**Pull request closed without merging**